server/routes/products.js

```diff
@@ -0,0 +1,82 @@
+const express = require('express');
+const router = express.Router();
+const Product = require('../models/product')
+
+// Get all products
+router.get('/', async (req, res) => {
+  try {
+    const products = await Product.find()
+    res.json(products)
+  } catch(err) {
+    res.status(500).json({ message: err.message })
+  }
+})
+
+// Get one product
+router.get('/:id', getProduct, (req, res) => {
+  res.send(res.product)
+})
+
+// Creating a product
+router.post('/', async (req, res) => {
+  const product = new Product({
+    name: req.body.name,
+    price: req.body.price,
+    img: req.body.img
+  })
+
+  try {
+    const newProduct = await product.save()
+    res.status(201).json(newProduct)
+  } catch(err) {
+    res.status(400).json({ message: err.message })
+  }
+})
+
+// Updating a product
+router.patch('/:id',  getProduct, async (req, res) => {
+  if(req.body.name != null) {
+    res.product.name = req.body.name
+  }
+  if(req.body.price != null) {
+    res.product.price = req.body.price
+  }
+  if(req.body.img != null) {
+    res.product.img = req.body.img
+  }
+  try {
+    const updatedProduct = await res.product.save()
```

Comment:
`.save()` is deprecated.

https://docs.mongodb.com/manual/reference/method/db.collection.save/

You should use `.insertOne` or `.replaceOne` instead